ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCaseBuilderUtil.java

```diff
@@ -147,12 +148,27 @@ private static Topic createTopicFromStatement(
       final CreateSource statement = (CreateSource) stmt.getStatement();
       final CreateSourceProperties props = statement.getProperties();
 
-      final FormatInfo valueFormatInfo = SourcePropertiesUtil.getValueFormat(props);
-      final Format valueFormat = FormatFactory.fromName(valueFormatInfo.getFormat());
+      final LogicalSchema logicalSchema = statement.getElements().toLogicalSchema();
+
+      final FormatInfo keyFormatInfo = SourcePropertiesUtil.getKeyFormat(props);
+      final Format keyFormat = FormatFactory.fromName(keyFormatInfo.getFormat());
+      final SerdeFeatures keySerdeFeats = buildKeyFeatures(
+          keyFormat, logicalSchema);
+
+      final Optional<ParsedSchema> keySchema =
+          keyFormat.supportsFeature(SerdeFeature.SCHEMA_INFERENCE)
+              ? buildValueSchema(sql, logicalSchema.key(), keyFormatInfo, keyFormat, keySerdeFeats)
```

Comment:
Rename method from `buildValueSchema` to `buildSchema` to avoid confusion? (And rename the params in the method as well)